scaladoc/src/dotty/tools/scaladoc/site/StaticSiteLoader.scala

```diff
@@ -0,0 +1,201 @@
+        val entries = Sidebar.load(path.toFile)
+        loadBasedOnYaml(entries)
+      }
+  }
+
+  def loadBasedOnYaml(yamlRoot: Sidebar.Root): StaticSiteRoot = {
+    val rootDest = ctx.docsPath.resolve("index.html").toFile
+    val rootIndex = yamlRoot.index
+      .map(Paths.get(root.getPath, _).toFile)
+      .filter(_.exists)
+      .fold(emptyTemplate(rootDest, "index")) { f =>
+        val loaded = loadTemplateFile(f)
+        if loaded.title.name != "index"
+        then report.warn("Property `title` will be overridden by project name", f)
+        loaded
+      }.copy(title = TemplateName.FilenameDefined(args.name))
+
+    def loadChild(pathFromRoot: Path): Sidebar => LoadedTemplate = {
+      case Sidebar.Category(optionTitle, optionIndexPath, nested, dir) =>
+        val indexPageOpt = optionIndexPath
+          .map(relativizeIfNeeded)
+          .map(_.toFile)
+          .filter(_.exists)
+          .map(loadTemplateFile)
+        val title = (
+          optionTitle.map(TemplateName.SidebarDefined(_)) ++
+          indexPageOpt.map(_.title)
+        ).headOption.getOrElse {
+          report.error(s"Title for subsection needs to be set in YAML config or in index file")
+          TemplateName.FilenameDefined("unnamed_section")
+        }
+        val categoryPath = dir.fold(pathFromRoot.resolve(toKebabCase(title.name)))(pathFromRoot.resolve(_))
+        val indexPage = indexPageOpt.getOrElse(emptyTemplate(categoryPath.resolve("index.html").toFile, title.name))
+
+        val children = optionIndexPath.filter(_ => nested.isEmpty).fold(
+          nested.map(child => loadChild(categoryPath)(child))
+        ) { indexPath =>
+          val indexPathDirectory = Paths.get(indexPath).getParent
+          val filesInDirectory = Option(root.toPath.resolve(indexPathDirectory).toFile.listFiles)
+          filesInDirectory.fold(List.empty) { files =>
+            val mappingFunc: File => File = file => {
+              val relativeFile = root.toPath.resolve(indexPathDirectory).relativize(file.toPath)
+              categoryPath.resolve(relativeFile).toFile
+            }
+            files.toList
+              .filter(_.toPath != indexPage.file.toPath)
+              .flatMap(file => loadRecursively(file, mappingFunc))
+          }
+        }
+
+        LoadedTemplate(indexPage, children, categoryPath.resolve("index.html").toFile)
+      case Sidebar.Page(optionTitle, pagePath) =>
+        val path = relativizeIfNeeded(pagePath)
+        val file = path.toFile
+        val templateFile = loadTemplateFile(file)
+        val withUpdatedTitle = optionTitle.fold(templateFile) { t => templateFile.title match
+          case _: TemplateName.FilenameDefined => templateFile.copy(title = TemplateName.SidebarDefined(t))
+          case _ => templateFile
+        }
+        LoadedTemplate(withUpdatedTitle, List.empty, pathFromRoot.resolve(file.getName).toFile)
+      case Sidebar.Root(_, _) =>
+        // Cannot happen
+        ???
```

Comment:
Maybe we could log it instead of throwing `NotImplementedException` even though it will never hit this branch